url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/client/SOCPlayerClient.java

```diff
@@ -0,0 +1,5459 @@
+        {
+            String name = cmd.substring(10).trim();
+            pi.getBoardPanel().setOtherPlayer(ga.getPlayer(name));
+            pi.getBoardPanel().setMode(SOCBoardPanel.CONSIDER_LT_CITY);
+
+            return true;
+        }
+        else
+        {
+            return false;
+        }
+    }
+
+    /**
+     * @return true if name is on the ignore list
+     */
+    protected boolean onIgnoreList(String name)
+    {
+        boolean result = false;
+        Enumeration ienum = ignoreList.elements();
+
+        while (ienum.hasMoreElements())
+        {
+            String s = (String) ienum.nextElement();
+            if (s.equals(name))
+            {
+                result = true;
+
+                break;
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * add this name to the ignore list
+     *
+     * @param name the name to add
+     */
+    protected void addToIgnoreList(String name)
+    {
+        name = name.trim();
+
+        if (!onIgnoreList(name))
+        {
+            ignoreList.addElement(name);
+        }
+    }
+
+    /**
+     * remove this name from the ignore list
+     *
+     * @param name  the name to remove
+     */
+    protected void removeFromIgnoreList(String name)
+    {
+        name = name.trim();
+        ignoreList.removeElement(name);
+    }
+
+    /** Print the current chat ignorelist in a channel. */
+    protected void printIgnoreList(ChannelFrame fr)
+    {
+        fr.print("* Ignore list:");
+
+        Enumeration ienum = ignoreList.elements();
+
+        while (ienum.hasMoreElements())
+        {
+            String s = (String) ienum.nextElement();
+            fr.print("* " + s);
+        }
+    }
+
+    /** Print the current chat ignorelist in a playerinterface. */
+    protected void printIgnoreList(SOCPlayerInterface pi)
+    {
+        pi.print("* Ignore list:");
+
+        Enumeration ienum = ignoreList.elements();
+
+        while (ienum.hasMoreElements())
+        {
+            String s = (String) ienum.nextElement();
+            pi.print("* " + s);
+        }        
+    }
+
+    /**
+     * send a command to the server with a message
+     * asking a robot to show the debug info for
+     * a possible move after a move has been made
+     *
+     * @param ga  the game
+     * @param pname  the robot name
+     * @param piece  the piece to consider
+     */
+    public void considerMove(SOCGame ga, String pname, SOCPlayingPiece piece)
+    {
+        String msg = pname + ":consider-move ";
+
+        switch (piece.getType())
+        {
+        case SOCPlayingPiece.SETTLEMENT:
+            msg += "settlement";
+
+            break;
+
+        case SOCPlayingPiece.ROAD:
+            msg += "road";
+
+            break;
+
+        case SOCPlayingPiece.CITY:
+            msg += "city";
+
+            break;
+        }
+
+        msg += (" " + piece.getCoordinates());
+        put(SOCGameTextMsg.toCmd(ga.getName(), nickname, msg), ga.isLocal);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 4683:4704, 4718:4739.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.